File: app/src/main/java/jp/ac/dendai/c/jtp/adventuresample/Music.java
package jp.ac.dendai.c.jtp.adventuresample;

import android.app.Activity;
import android.media.MediaPlayer;

import jp.ac.dendai.c.jtp.adventuresample.scene.Scene;

/**
 * Created by yuuki on 2017/05/18.
 */

public class Music {
    public MediaPlayer mp = new MediaPlayer();

    public void MusicStop(){
        mp.stop();
        mp.reset();
        // リソースの解放
        mp.release();
    }

    public void music(Activity activity, Scene scene){
        if(mp != null) {
            MusicStop();
        }
        mp = MediaPlayer.create(activity, scene.getMusicId());
        mp.start();
    }
}

File: app/src/main/java/jp/ac/dendai/c/jtp/adventuresample/Music.kt
package jp.ac.dendai.c.jtp.adventuresample

import android.app.Activity
import android.media.MediaPlayer

import jp.ac.dendai.c.jtp.adventuresample.scene.Scene

/**
 * Created by yuuki on 2017/05/18.
 */

class Music {
    var activity: Activity? = null
    var scene: Scene? = null

    var mp: MediaPlayer? = MediaPlayer()

    fun MusicStart() {
        mp = MediaPlayer.create(activity, scene!!.musicId)
        mp!!.start()
    }

    fun MusicStop() {
        mp!!.stop()
        mp!!.reset()
        // リソースの解放
        mp!!.release()
    }

    fun music(activity: Activity, scene: Scene) {
        this.activity = activity
        this.scene = scene

        if (mp != null) {
            MusicStop()
        }
        MusicStart()
    }
}
